src/frequenz/repo/config/version.py

```diff
@@ -326,6 +326,64 @@ def branches(self) -> dict[str, BranchVersion]:
         """
         return self._branches
 
+    def find_last_tag(self) -> semver.Version | None:
+        """Find the last tag.
+
+        Returns:
+            If we are at a tag, return the [current
+                tag][frequenz.repo.config.version.RepoVersionInfo.current_tag]. If we
+                are at a branch, return the last tag matching the branch major and minor
+                (if any). If there are no matching tags, return `None`.
+        """
+        if self._current_tag:
+            return self._current_tag
+        branch = self.current_branch
+        if branch is None:
+            return None
+        tags = [t for t in self._tags.values() if t.major == branch.major]
+        if branch.minor is not None:
+            tags = [t for t in tags if t.minor == branch.minor]
+        if not tags:
+            return None
+        return max(tags)
```

Comment:
or the exceedingly beautiful `return tags and max(tags)`

Reply:
I don't see a clear winner here, so I will leave it as it is unless someone really thinks it improves readability considerably to change it. BTW @shsms, your code is not equivalent, because if tags is `[]` it will return `[]` instead of `None `.

Reply:
ah traps

Reply:
I made this comment not only to improve readability but consistency as the function mainly uses the pathern `if class is not None` and here was `if not ...`. Anyway not a big winner...